src/LSP12/authenticateIssuedAssets/authenticateIssuedAssets.test.ts

```diff
@@ -0,0 +1,260 @@
+import { INTERFACE_IDS } from '@lukso/lsp-smart-contracts';
+import { Signer, keccak256, toUtf8Bytes } from 'ethers';
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+
+// types
+import {
+    LSP7Mintable__factory,
+    LSP7Mintable,
+    UniversalProfile__factory,
+    UniversalProfile,
+} from '../../typechain';
+
+// util
+import {
+    IssuerAssets,
+    addDigitalAssetCreators,
+    addIssuedAssets,
+    authenticateIssuedAssets,
+    removeDigitalAssetCreators,
+    removeIssuedAssets,
+} from '../..';
+
+describe('authenticateIssuedAssets', () => {
+    let context: {
+        creatorsOwner: Signer;
+        firstCreator: UniversalProfile;
+        secondCreator: UniversalProfile;
+        digitalAssetsOwner: Signer;
+        firstDigitalAsset: LSP7Mintable;
+        secondDigitalAsset: LSP7Mintable;
+        thirdDigitalAsset: LSP7Mintable;
+    };
+    before('deploy UP and token contract', async () => {
+        const signers = await ethers.getSigners();
+        const creatorsOwner = signers[0];
+        const digitalAssetsOwner = signers[1];
+
+        const firstCreator = await new UniversalProfile__factory(creatorsOwner).deploy(
+            creatorsOwner.address,
+        );
+        const secondCreator = await new UniversalProfile__factory(creatorsOwner).deploy(
+            creatorsOwner.address,
+        );
+
+        const firstDigitalAsset = await new LSP7Mintable__factory(digitalAssetsOwner).deploy(
+            'FirstTestToken',
+            'FTT',
+            digitalAssetsOwner.address,
+            true,
+        );
+        const secondDigitalAsset = await new LSP7Mintable__factory(digitalAssetsOwner).deploy(
+            'SecondTestToken',
+            'STT',
+            digitalAssetsOwner.address,
+            true,
+        );
+        const thirdDigitalAsset = await new LSP7Mintable__factory(digitalAssetsOwner).deploy(
+            'ThirdTestToken',
+            'TTT',
+            digitalAssetsOwner.address,
+            true,
+        );
+
+        context = {
+            creatorsOwner,
+            firstCreator,
+            secondCreator,
+            digitalAssetsOwner,
+            firstDigitalAsset,
+            secondDigitalAsset,
+            thirdDigitalAsset,
+        };
+    });
+
+    it('should throw when token address is not hex', async () => {
+        const issuerAddress = 'address';
+
+        await expect(authenticateIssuedAssets(issuerAddress, ethers.provider)).to.be.rejectedWith(
+            `The parameter \`issuerAddress\` is not a valid address nor a valid contract instance of \`ERC725Y\`. Value: '${issuerAddress}'`,
+        );
+    });
+
+    it('should throw when token address is hex with less than 20 bytes', async () => {
+        const issuerAddress = keccak256(toUtf8Bytes('address')).substring(0, 40);
```

Comment:
Here there is only 19 bytes, but I think it would read better as plain address literal with a simple comment. Here you have to figure out what the whole line does.

```suggestion
        // invalid address that is only 19 bytes long
        const issuerAddress = "0x421683f821a0574472445355be6d2b769119e8"
```